test_conformance/api/test_semaphores.cpp

```diff
@@ -0,0 +1,1137 @@
+        }                                                                      \
+        if (size != sizeof(value))                                             \
+        {                                                                      \
+            test_fail(                                                         \
+                "ERROR: Returned size of parameter %s does not validate! "     \
+                "(expected %d, got %d)\n",                                     \
+                #param_name, (int)sizeof(value), (int)size);                   \
+        }                                                                      \
+    } while (false)
+
+#define SEMAPHORE_PARAM_TEST_ARRAY(param_name, param_type, num_params,         \
+                                   expected)                                   \
+    do                                                                         \
+    {                                                                          \
+        param_type value[num_params];                                          \
+        size_t size;                                                           \
+        cl_int error = clGetSemaphoreInfoKHR(sema, param_name, sizeof(value),  \
+                                             &value, &size);                   \
+        test_error(error, "Unable to get " #param_name " from semaphore");     \
+        if (size != sizeof(value))                                             \
+        {                                                                      \
+            test_fail(                                                         \
+                "ERROR: Returned size of parameter %s does not validate! "     \
+                "(expected %d, got %d)\n",                                     \
+                #param_name, (int)sizeof(value), (int)size);                   \
+        }                                                                      \
+        if (memcmp(value, expected, size) != 0)                                \
+        {                                                                      \
+            test_fail("ERROR: Parameter %s did not validate!\n", #param_name); \
+        }                                                                      \
+    } while (false)
+
+static const char* source = "__kernel void empty() {}";
+
+// Helper function that signals and waits on semaphore across two different
+// queues.
+static int semaphore_cross_queue_helper(cl_device_id deviceID,
+                                        cl_context context,
+                                        cl_command_queue queue_1,
+                                        cl_command_queue queue_2)
+{
+    cl_int err;
+
+    if (!is_extension_available(deviceID, "cl_khr_semaphore"))
+    {
+        log_info("cl_khr_semaphore is not supported on this platoform. "
+                 "Skipping test.\n");
+        return TEST_SKIP;
```

Comment:
change to  "return TEST_SKIPPED_ITSELF".  Using TEST_SKIP here causes the test to return false  failures on implementations that do not support cl_khr_semaphore.

Reply:
Ack